sql/core/src/main/scala/org/apache/spark/sql/execution/CacheManager.scala

```diff
@@ -46,38 +45,17 @@ case class CachedData(plan: LogicalPlan, cachedRepresentation: InMemoryRelation)
  */
 class CacheManager extends Logging {
 
-  @transient
-  private val cachedData = new java.util.LinkedList[CachedData]
-
-  @transient
-  private val cacheLock = new ReentrantReadWriteLock
-
-  /** Acquires a read lock on the cache for the duration of `f`. */
-  private def readLock[A](f: => A): A = {
-    val lock = cacheLock.readLock()
-    lock.lock()
-    try f finally {
-      lock.unlock()
-    }
-  }
-
-  /** Acquires a write lock on the cache for the duration of `f`. */
-  private def writeLock[A](f: => A): A = {
-    val lock = cacheLock.writeLock()
-    lock.lock()
-    try f finally {
-      lock.unlock()
-    }
-  }
+  @transient @volatile
+  private var cachedData = IndexedSeq[CachedData]()
```

Comment:
we'd better add some comments to explain the access pattern of `cachedData`. e.g. adding/removing elements should be done by creating a new seq, and locked by this.